pyseir/rt/infer_rt.py

```diff
@@ -0,0 +1,617 @@
+           in R so that this process does not move argmax (the peak in probability)
+        """
+        # TODO FOR ALEX: Please expand this and describe more clearly these cutoffs
+        use_sigma = (
+            min(
+                self.max_scaling_sigma,
+                max(1.0, math.sqrt(self.scale_sigma_from_count / timeseries_scale)),
+            )
+            * self.default_process_sigma
+        )
+
+        process_matrix = sps.norm(loc=self.r_list, scale=use_sigma).pdf(self.r_list[:, None])
+
+        # process_matrix applies gaussian smoothing to the previous posterior to make the prior.
+        # But when the gaussian is wide much of its distribution function can be outside of the
+        # range Reff = (0,10). When this happens the smoothing is not symmetric in R space. For
+        # R<1, when posteriors[previous_day]).argmax() < 50, this asymmetry can push the argmax of
+        # the prior >10 Reff bins (delta R = .2) on each new day. This was a large systematic error.
+
+        # Ensure smoothing window is symmetric in X direction around diagonal
+        # to avoid systematic drift towards middle (Reff = 5). This is done by
+        # ensuring the following matrix values are 0:
+        # 1 0 0 0 0 0 ... 0 0 0 0 0 0
+        # * * * 0 0 0 ... 0 0 0 0 0 0
+        # ...
+        # * * * * * * ... * * * * 0 0
+        # * * * * * * ... * * * * * *
+        # 0 0 * * * * ... * * * * * *
+        # ...
+        # 0 0 0 0 0 0 ... 0 0 0 * * *
+        # 0 0 0 0 0 0 ... 0 0 0 0 0 1
+        sz = len(self.r_list)
+        for row in range(0, sz):
+            if row < (sz - 1) / 2:
+                process_matrix[row, 2 * row + 1 : sz] = 0.0
+            elif row > (sz - 1) / 2:
+                process_matrix[row, 0 : sz - 2 * (sz - row)] = 0.0
+
+        # (3a) Normalize all rows to sum to 1
+        row_sums = process_matrix.sum(axis=1)
+        for row in range(0, sz):
+            process_matrix[row] = process_matrix[row] / row_sums[row]
+
+        return use_sigma, process_matrix
+
+    def get_posteriors(self, timeseries_type, plot=False):
+        """
+        Generate posteriors for R_t.
+
+        Parameters
+        ----------
+        ----------
+        timeseries_type: TimeseriesType
+            New X per day (cases, deaths etc).
+        plot: bool
+            If True, plot a cool looking est of posteriors.
+
+        Returns
+        -------
+        dates: array-like
+            Input data over a subset of indices available after windowing.
+        times: array-like
+            Output integers since the reference date.
+        posteriors: pd.DataFrame
+            Posterior estimates for each timestamp with non-zero data.
+        start_idx: int
+            Index of first Rt value calculated from input data series
+            #TODO figure out why this value sometimes truncates the series
+            
+        """
+        dates, timeseries = self.get_timeseries(timeseries_type=timeseries_type)
```

Comment:
Could you pass in dates and timeseries rather than access it in this function? Makes testing this function easier because we can play around with the timeseries without having to rely on the class